tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -2854,6 +2886,38 @@ tensorflow::Status ConvertSegmentToGraphDef(
   return tensorflow::Status::OK();
 }
 
+bool InputEdgeValidator::operator()(const tensorflow::Edge* in_edge) const {
+  if (in_edge->IsControlEdge()) return true;
+  PartialTensorShape shape;
+  tensorflow::DataType dtype;
+  GetInputProperties(graph_properties_, in_edge->src(), in_edge->src_output(),
+                     &shape, &dtype);
+  nvinfer1::DataType trt_dtype;
+  Status status = ValidateInputProperties(shape, dtype, &trt_dtype);
+  if (!status.ok()) {
+    VLOG(2) << "--> Need to remove input node " << in_edge->dst()->name()
+            << ": " << status;
+    return false;
+  }
+  if (shape.dims() < 3 && in_edge->src()->type_string() != "Const") {
```

Comment:
shape.dims() < 3?
Not quite sure about the check.

TRT 3.0 requires input to be of rank 4 (batch + 3 dimension). This requirement has been removed in TRT 4.0.
If this check is related to that, we need to add a compile guard.

Reply:
As discussed offline, I'll leave this as is since there are more places to change besides this one regarding the 3 dims requirement. And we can issue another PR to fix the problem.